src/sources/socket/mod.rs

```diff
@@ -142,44 +158,40 @@ impl SourceConfig for SocketConfig {
                 )
             }
             Mode::Udp(config) => {
-                let host_key = config
-                    .host_key()
-                    .clone()
-                    .unwrap_or_else(|| log_schema().host_key().to_string());
+                let log_namespace = cx.log_namespace(config.log_namespace);
                 let decoder = DecodingConfig::new(
                     config.framing().clone(),
                     config.decoding().clone(),
                     LogNamespace::Legacy,
                 )
                 .build();
-                Ok(udp::udp(config, host_key, decoder, cx.shutdown, cx.out))
+                Ok(udp::udp(
+                    config,
+                    decoder,
+                    cx.shutdown,
+                    cx.out,
+                    log_namespace,
+                ))
             }
             #[cfg(unix)]
             Mode::UnixDatagram(config) => {
-                let host_key = config
-                    .host_key
-                    .unwrap_or_else(|| log_schema().host_key().to_string());
                 let decoder = DecodingConfig::new(
-                    config.framing.unwrap_or_else(default_framing_message_based),
+                    config
+                        .clone()
+                        .framing
+                        .unwrap_or_else(default_framing_message_based),
                     config.decoding.clone(),
                     LogNamespace::Legacy,
                 )
                 .build();
-                unix::unix_datagram(
-                    config.path,
-                    config.socket_file_mode,
-                    config
-                        .max_length
-                        .unwrap_or_else(crate::serde::default_max_length),
-                    host_key,
-                    decoder,
-                    cx.shutdown,
-                    cx.out,
-                )
+
+                let log_namespace = cx.log_namespace(config.log_namespace);
+
+                unix::unix_datagram(config, decoder, cx.shutdown, cx.out, log_namespace)
```

Comment:
:thought_balloon:  Looks like some additional `clone()`s are being introduced as a result of moving ownership of the `config` to the `unix_<...>()` functions, which isn't ideal if it can be avoided, but since this seems to only be done once during initialization I guess it's alright.

Reply:
I changed the logic slightly so that the whole config isn't cloned above, just the specific fields that we need to extract.